File: SmartAIServer/routes/sessions.js
import { Router } from 'express';
import { stmts }  from '../db.js';

const router = Router();

// GET /api/sessions/:id/history
router.get('/:id/history', (req, res) => {
  const messages = stmts.getHistory.all(req.params.id);
  res.json({
    messages: messages.map(m => ({
      id:        m.id,
      role:      m.role,
      content:   m.content,
      timestamp: m.timestamp,
    })),
  });
});

// DELETE /api/sessions/:id/history
router.delete('/:id/history', (req, res) => {
  stmts.clearHistory.run(req.params.id);
  res.json({ cleared: true });
});

export default router;

File: SmartAIServer/routes/sessions.py
"""Chat session history endpoints."""
import logging

from fastapi import APIRouter

import db

logger = logging.getLogger("smartai.sessions")

router = APIRouter()


@router.get("/{session_id}/history")
async def get_history(session_id: str):
    messages = db.get_history(session_id)
    return {
        "messages": [
            {"id": m["id"], "role": m["role"], "content": m["content"], "timestamp": m["timestamp"]}
            for m in messages
        ]
    }


@router.delete("/{session_id}/history")
async def clear_history(session_id: str):
    db.clear_history(session_id)
    logger.info("Cleared history for session %s", session_id[:8])
    return {"cleared": True}
